src/LoginProviders/LogLoginAttemptsTrait.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Rhubarb\Scaffolds\Authentication\LoginProviders;
+
+use Rhubarb\Crown\LoginProviders\Exceptions\LoginDisabledException;
+use Rhubarb\Crown\LoginProviders\Exceptions\LoginDisabledFailedAttemptsException;
+use Rhubarb\Crown\LoginProviders\Exceptions\LoginExpiredException;
+use Rhubarb\Crown\LoginProviders\Exceptions\LoginFailedException;
+use Rhubarb\Scaffolds\Authentication\UserLoginAttempt;
+
+trait LogLoginAttemptsTrait
+{
+    public function login($username, $password)
+    {
+        try {
+            $loginStatus = parent::login($username, $password);
+
+            if ($loginStatus) {
+                $this->createSuccessfulUserLoginAttempt($username);
+            }
+
+            return $loginStatus;
+        } catch (LoginDisabledException $loginDisabledException) {
+            $this->createFailedUserLoginAttempt($username, (string) $loginDisabledException);
+            throw $loginDisabledException;
+        } catch (LoginFailedException $loginFailedException) {
+            $this->createFailedUserLoginAttempt($username, (string) $loginFailedException);
+            throw $loginFailedException;
+        } catch (LoginExpiredException $loginExpiredException) {
```

Comment:
Hi Michael, I'm missing something - these exceptions don't exist on this branch and I'd expect to see something in ModelLoginProvider/LoginProvider that throws them too... missing commits?

Reply:
Hi Bert 
These exceptions are defined within the main Rhubarb module I just followed the same pattern and have a separate PR in for the Rhubarb module that includes these new Exception. 

Please see link below:
https://github.com/RhubarbPHP/Rhubarb/pull/55